### PR TITLE
add json serializable IP type to pkg/api/unversioned

### PR DESCRIPTION
**Pull request by mikedanese**
cc @lavalamp @bgrant0607

### CODE REVIEW

**Comment by thockin** [comment]
Maybe it's because I was staring at validation code recently, but I have begun to think we should not be doing this, and instead validating it all in one place.  This will just give an error message that is not consistent with our other error messages.

File: pkg/api/unversioned/ip_test.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+type IPHolder struct {
+	I IP `json:"ip,omitempty"`
+}
+
+func TestIPMarshalYAML(t *testing.T) {
+	cases := []struct {
+		input  IP
+		result string
+	}{}
```
Comment (caesarxuchao) [inline pkg/api/unversioned/ip_test.go]: empty test cases?

**Comment by caesarxuchao** [comment]
I thought we agreed to add it into the componentconfig group. I think you should add it in pkg/apis/componentconfig/types.go, and register the same struct in both pkg/apis/componentconfig/reigster.go and pkg/apis/componentconfig/v1alpha1/reigster.go.

File: pkg/api/unversioned/ip.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+func (ip *IP) UnmarshalJSON(b []byte) error {
+	var str string
+	err := json.Unmarshal(b, &str)
+	if err != nil {
+		return err
+	}
+
+	pip := net.ParseIP(str)
+	if pip == nil {
+		return fmt.Errorf("%q is not a valid representation of an IP address", str)
```
Comment (caesarxuchao) [inline pkg/api/unversioned/ip.go]: @thockin , do you mean move this check to pkg/apis/componentconfig/validation? That will lose the `str`, so I think the current way is better.

Comment (thockin) [inline pkg/api/unversioned/ip.go]: I meant to hold it in a string and validate it in validation.go.

**Pull request closed without merging**